codec-native-quic/src/main/c/netty_quic_boringssl.c

```diff
@@ -597,12 +612,59 @@ void keylog_callback(const SSL* ssl, const char* line) {
     (*e)->CallVoidMethod(e, keylogCallback, keylogCallbackMethod, (jlong) ssl, keyString);
 }
 
-static jlong netty_boringssl_SSLContext_new0(JNIEnv* env, jclass clazz, jboolean server, jbyteArray alpn_protos, jobject handshakeCompleteCallback, jobject certificateCallback, jobject verifyCallback, jobject servernameCallback, jobject keylogCallback, jint verifyMode, jobjectArray subjectNames) {
+int new_session_callback(SSL *ssl, SSL_SESSION *session) {
+    SSL_CTX* ctx = SSL_get_SSL_CTX(ssl);
+    if (ctx == NULL) {
+        return 0;
+    }
+
+    JNIEnv* e = NULL;
+    if (quic_get_java_env(&e) != JNI_OK) {
+        return 0;
+    }
+
+    jobject sessionCallback = SSL_CTX_get_ex_data(ctx, sessionCallbackIdx);
+    if (sessionCallback == NULL) {
+        return 0;
+    }
+
+    uint8_t *session_data = NULL;
+    size_t session_data_len = 0;
+    if (SSL_SESSION_to_bytes(session, &session_data, &session_data_len) == 0) {
+        // Get session error
+        return 0;
+    }
+
+    jbyteArray sessionBytes = to_byte_array(e, session_data, session_data_len);
+    OPENSSL_free((void *)session_data);
```

Comment:
Add a link to https://commondatastorage.googleapis.com/chromium-boringssl-docs/ssl.h.html#SSL_SESSION_to_bytes to explain why we need to call free